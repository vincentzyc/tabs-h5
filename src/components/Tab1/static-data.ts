export const expensesText = `【温馨提示】
1.联通大王卡月费：29元/月
2.套内资费∶联通大王卡套餐内包含30GB定向专属流量、免费接听、全国无漫游费。
3.套外资费∶套餐外流量5 元/GB/月。
4.亲情号特权：用户初始可添加号码为3个，会员等级每提高1级，最多可添加的号码增加1个（最多可添加8个亲情号）。
5.当月套餐上网流量累计使用达200GB时，系统将关闭上网功能，次月自动打开。王上可通过"王卡助手公众号"的"特权-个人中心"菜单办理"解除流量封顶"业务。如当月解除封顶，超出部分按照套餐外流量收费。
6.入网当月及次月内不能订购或点播SP业务。
7.目前王卡资费优惠均不包含国际地区、港澳台地区。

更多详情请咨询当地10016。
客服电话：4008168562
`