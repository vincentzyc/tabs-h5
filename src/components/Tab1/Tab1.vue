<template>
  <SelectNumber />
  <img src="@/assets/img/tab1-2.jpg" alt="产品说明" class="widthfull mg-t10" />
  <img src="@/assets/img/tab1-3.jpg" alt="产品说明" class="widthfull" />
  <img src="@/assets/img/tab1-4.png" alt="产品说明" class="widthfull" />
  <div class="text-center expenses-wrap">
    <span class="expenses-text" @click="showExpenses()">资费详情说明</span>
  </div>
  <van-popup class="agreement-popup" v-model:show="show" round>
    <div class="agreement-wrap">{{ expensesText }}</div>
    <div class="agreement-confirm" @click="close()">我知道了</div>
  </van-popup>
</template>

<script setup lang="ts">
import { expensesText } from "./static-data";

let show = $ref(false);
function showExpenses() {
  show = true;
}
function close() {
  show = false;
}
</script>
