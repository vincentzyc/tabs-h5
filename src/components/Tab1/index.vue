<template>
  <div>联通</div>
  <img src="@/assets/img/tab1-2.jpg" alt="产品说明" class="widthfull" />
  <img src="@/assets/img/tab1-3.jpg" alt="产品说明" class="widthfull" />
  <img src="@/assets/img/tab1-4.png" alt="产品说明" class="widthfull" />
  <div class="text-center expenses-wrap">
    <span class="expenses-text" @click="showExpenses()">资费详情说明</span>
  </div>
  <van-popup class="center-popup" v-model:show="show" round>
    <div class="popup-wrap">{{ expensesText }}</div>
  </van-popup>
</template>

<script setup lang="ts">
import { expensesText } from "./static-data";
let show = $ref(false);
function showExpenses() {
  show = true;
}
</script>
