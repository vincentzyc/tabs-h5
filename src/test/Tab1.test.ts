
import { mount } from '@vue/test-utils'
import { Lazyload } from 'vant';
import { createTestingPinia } from '@pinia/testing'
import Tab1 from '@/components/Tab1/Tab1.vue'

test('mount Tab1 component', async () => {
  expect(Tab1).toBeTruthy()

  const wrapper = mount(Tab1, {
    global: {
      plugins: [createTestingPinia()],
      directives: {
        lazy: Lazyload
      }
    },
  })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('.expenses-text').trigger('click')

  await nextTick()

  expect(wrapper.find('.agreement-popup').exists()).toBe(true)
})
