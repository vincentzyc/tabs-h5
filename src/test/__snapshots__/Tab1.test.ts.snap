// Vitest Snapshot v1

exports[`mount Tab1 component 1`] = `
"<!-- <SelectNumber @selected=\\"selectNum\\" showBelongCity showSearchNumber showNumberRule /> -->
<div class=\\"marquee-single flex\\">
  <ul class=\\"scroll-wrapper\\">
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">李**抢得靓号 1305678****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">蔡**抢得靓号 1308288****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">张**抢得靓号 1326668****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">罗**抢得靓号 1665688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">林**抢得靓号 1556688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">周**抢得靓号 1306566****</span></li>
  </ul>
</div>
<!-- <img src=\\"@/assets/img/tab1-2.jpg\\" alt=\\"产品说明\\" class=\\"widthfull mg-t10\\" />
  <img src=\\"@/assets/img/tab1-3.jpg\\" alt=\\"产品说明\\" class=\\"widthfull\\" />
  <img alt=\\"产品说明\\" class=\\"widthfull img4\\" v-lazy=\\"img4\\" />
  <div class=\\"text-center expenses-wrap\\">
    <span class=\\"expenses-text\\" @click=\\"showExpenses()\\">资费详情说明</span>
  </div>
  <van-popup v-model:show=\\"showForm\\" class=\\"common-popup\\" style=\\"max-width: 576px\\" teleport=\\"body\\">
    <div class=\\"common-wrap\\" v-if=\\"selectNumItem\\">
      <p class=\\"info-text-number\\"
        >已选择<span class=\\"info-number\\">靓号 {{ selectNumItem.num }}</span> {{ showCity }}</p
      >
      <p class=\\"info-text\\">根据国家手机号卡实名制规定</p>
      <p class=\\"info-text\\">请如实填写信息，以便我们及时为您送达</p>
      <BaseForm :handleNo=\\"selectNumItem.num\\" @submit=\\"submitOrder\\" />
      <Agreement title=\\"填写并提交视为阅读并同意\\" :agrList=\\"agrList\\" v-model:checked=\\"checked\\" />
    </div>
  </van-popup>
  <van-popup class=\\"agreement-popup\\" v-model:show=\\"show\\" round>
    <div class=\\"agreement-wrap\\">{{ expensesText }}</div>
    <div class=\\"agreement-confirm\\" @click=\\"close()\\">我知道了</div>
  </van-popup> -->"
`;
